js/classes.js

```diff
@@ -96,8 +96,35 @@ class Projectile {
     this.position.x += this.velocity.x;
     this.position.y += this.velocity.y;
   }
+  static createProjectile(xVelocity = 0, yVelocity = -10, color = "red") {
+    return new Projectile({
+        position: {
+          x: player.position.x + player.width / 2,
+          y: player.position.y,
+        },
+        velocity: {
+          x: xVelocity,
+          y: yVelocity,
+        },
+      color: color,
+      })
+}
+}  
+   function playerShoots() {
+   if((keys.space.pressed && !game.over) || (keys.mouse.pressed && !game.over)) {
```

Comment:
1) Fix indent please 😭😭
2) `playerShoots()`  should not include the '**keys**'  or '**game**' global variables / any of that logic

Reply:
Sorry about the indentation; I wrote some of it on my phone.